util/import.ts

```diff
@@ -0,0 +1,22 @@
+import { TestCaseInfo } from '../state/atoms/patientTestCase';
+
+export function bundleToTestCase(bundle: fhir4.Bundle): TestCaseInfo {
+  if (!bundle.entry || bundle.entry.length === 0) {
+    throw new Error('Bundle has no entries');
```

Comment:
When we are uploading more than one JSON file at a time, it would be useful to include the name of the file that produced the error. I tested a malformed JSON and it was nice that I was able to see which file produced the parsing error. With the non-Bundle and non-patient resource errors, we don't automatically know which JSON produced the error. This isn't a big deal when we are only uploading a few files, but when we have a lot of files it takes some time to compare the successful file names to the group of files we tried to upload and see which bundle threw the error.

I also noticed that for the other errors in the resource panel (like malformed JSON), if one of the bundles fails, the app will still attempt to import all the remaining bundles. However, for these errors, if the error is thrown, then none of the remaining JSON files will be imported. I tried importing 6 JSON files, where the fourth was missing bundle entries, and I got 3 success and 1 failure rather than 5 success and 1 failure. Should we try and change the logic here to reflect that? Maybe add "Success: __, Failed: __, Unattempted: ___" (or some other wording), or group them into the "Failed?"

Reply:
Oh shoot yeah. The errors that are thrown here get into the root `catch` block of our `Promise.all`. I talked to @elsaperelli about this, she's gonna add a try catch to make sure these specific errors fail safely as well. They should be grouped into failures